### ISSUE TITLE
Some calls to `eth_getTransactionReceipt` fail to deserialize

### ISSUE THREAD

**Issue by ebadiere** [hashgraph/hedera-json-rpc-relay#3062]
### Description

The following is logged when calling this endpoint `contracts/results/0x15784af816559267de17d367b86c93db6f3143df46b97258f4a6492a5ccf04e8` :
```
Cannot read properties of null (reading 'toString'
```
Calling that using postman or curl returns the response with no issues.  Somehow the relay is not handling the response correctly.

### Steps to reproduce

No reproducer yet.  This was identified by the alerts and in the logs.

### Hedera network

mainnet

### Version

v0.57.0

### Operating system

Linux

**Comment by natanasow** [comment]
@ebadiere 

It's not reproducible as you mentioned above. I did testing against the following relay's versions (v0.57, v0.56, v0.55). Also, I executed the same call against the ws-server. Every time the request succeeded without any error.

Do you have an idea of what else we can investigate? Or can we close the issue and mark it as done?

**Comment by ebadiere** [comment]
@natanasow 

It just happened again on testnet running `eth_getBlockByHash`.  Here is what I am seeing in the logs:

```
[2024-10-25 13:17:01.283 +0000] [34mDEBUG[39m (rpc-server/84 on testnet-hashio-767fc5577c-xrwfv): [36m[Request ID: 399491af-2e33-490c-9340-25be3f376220] Validating method parameters for eth_getBlockByHash, params: ["0x247d411acfd59a0b35008cdc65c8da8f9c073168523e88423001d7df71ea9095",true][39m...

[2024-10-25 13:17:01.283 +0000] [90mTRACE[39m (relay-eth/84 on testnet-hashio-767fc5577c-xrwfv): [36m[Request ID: 399491af-2e33-490c-9340-25be3f376220] getBlockByHash(hash=0x247d411acfd59a0b35008cdc65c8da8f9c073168523e88423001d7df71ea9095, showDetails=true)[39m....

[2024-10-25 13:17:01.286 +0000] [90mTRACE[39m (redisCache/84 on testnet-hashio-767fc5577c-xrwfv): [36m[Request ID: 399491af-2e33-490c-9340-25be3f376220] Returning cached value getBlock.0x247d411acfd59a0b35008cdc65c8da8f9c073168523e88423001d7df71ea9095:{"count":12,"hapi_version":"0.49.0","hash":"0x247d411acfd59a0b35008cdc65c8da8f9c073168523e88423001d7df71ea9095df34af1aba60dd34d77d494fec2518f9","name":"2024-04-25T11_18_52.450373578Z.rcd.gz","number":3288975,"previous_hash":"0x86d8811e026cda4afbcd9ae9d14a21760085a2bc56d74b166a70a1b1708bde923c4914a58cfe0f42914f6889a43bf9a4","size":5411,"timestamp":{"from":"1714043932.450373578","to":"1714043933.966640830"},"gas_used":0,"logs_bloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"} on blocks/ call[39m....

[36m[Request ID: 399491af-2e33-490c-9340-25be3f376220] Returning cached value getBlock.0x247d411acfd59a0b35008cdc65c8da8f9c073168523e88423001d7df71ea9095:{"count":12,"hapi_version":"0.49.0","hash":"0x247d411acfd59a0b35008cdc65c8da8f9c073168523e88423001d7df71ea9095df34af1aba60dd34d77d494fec2518f9","name":"2024-04-25T11_18_52.450373578Z.rcd.gz","number":3288975,"previous_hash":"0x86d8811e026cda4afbcd9ae9d14a21760085a2bc56d74b166a70a1b1708bde923c4914a58cfe0f42914f6889a43bf9a4","size":5411,"timestamp":{"from":"1714043932.450373578","to":"1714043933.966640830"},"gas_used":0,"logs_bloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"} on blocks/ call[39m....

[2024-10-25 13:17:01.295 +0000] [34mDEBUG[39m (mirror-node/84 on testnet-hashio-767fc5577c-xrwfv): [36m[Request ID: 399491af-2e33-490c-9340-25be3f376220] [GET] contracts/results/logs?timestamp=gte:1714043932.450373578&timestamp=lte:1714043933.966640830&limit=100&order=asc 200 4 ms[39m...

[2024-10-25 13:17:01.295 +0000] [31mERROR[39m (rpc-server/84 on testnet-hashio-767fc5577c-xrwfv): [36m[Request ID: 399491af-2e33-490c-9340-25be3f376220] Error invoking RPC: Cannot read properties of null (reading 'substring')[39m

[2024-10-25 13:17:01.295 +0000] [32mINFO[39m (rpc-server/84 on testnet-hashio-767fc5577c-xrwfv): [36m[Request ID: 399491af-2e33-490c-9340-25be3f376220] [POST]: eth_getBlockByHash 500 13 ms[39m
```
```

**Comment by natanasow** [comment]
@ebadiere Finally figured it out.

tl;dr; if we submit an ethereum transaction via relay with an invalid operator id, the `from` value returned from the mirror node is null but it must be the signer's address (derived by signature)

Synced up with the mirror node team and opened a ticket there https://github.com/hashgraph/hedera-mirror-node/issues/9667.

As a fix, we can check for a nullable `from` here https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/formatters.ts#L169 but that will resolve in ethereum inequivalence because in the ethereum ecosystem, there is no way to exist a transaction on-chain without `from` value.

cc: @Nana-EC because the task finalization might require interventions in services or mirror node

**Comment by natanasow** [comment]
WIll be resolved by https://github.com/hashgraph/hedera-json-rpc-relay/issues/3171

**Comment by natanasow** [comment]
@quiet-node Yes, it's fixed here https://github.com/hashgraph/hedera-json-rpc-relay/pull/3254. I'm closing this one.
